### ISSUE TITLE
`ignore` does not work correctly in 3.2.6

### ISSUE THREAD

**Issue by longlho** [mrmlnc/fast-glob#325]
### Environment

  * OS Version: Mac OSX
  * Node.js Version: 16

### Actual behavior

Should ignore files that match the pattern

### Expected behavior
Did not

### Steps to reproduce

Repro repo: https://github.com/andeliv/formatjs-cli-error

1. Install `fast-glob@3.2.6`
2. `cd packages/trn`
3. Run `node`
4. `require('fast-glob').sync('../**/src/**/*.{ts,tsx}', {ignore: '../**/src/**/*.{d,test}.{ts,tsx}'})`

Results came back as 
```js
[
  '../web-components/src/index.tsx',
  '../web-main/packages/web/src/@types/logger.d.ts',
  '../web-main/packages/web/src/components/messages.ts',
  '../web-main/packages/web/src/components/messages.tsx'
]
```

but it should have been

```js
[
  '../web-components/src/index.tsx',
  '../web-main/packages/web/src/components/messages.ts',
  '../web-main/packages/web/src/components/messages.tsx'
]
```

It worked fine in 3.2.5

**Comment by fjc0k** [comment]
+1. @mrmlnc 

```javascript
import globby from 'globby'

async function main() {
  console.log(
    await globby(['../../node_modules/globby/*.js', '!../../node_modules/globby/index.js'], {
      cwd: __dirname,
      onlyFiles: false,  
      dot: true,
      onlyFiles: false,
      gitignore: false,
      absolute: true,
    })
  )
}

main()
```

## 3.2.5: correct

![image](https://user-images.githubusercontent.com/13151189/124797641-13c11f00-df85-11eb-97cc-8020189eb109.png)
![image](https://user-images.githubusercontent.com/13151189/124797698-23406800-df85-11eb-8cc7-d1e32521cb64.png)

## 3.2.6: wrong

![image](https://user-images.githubusercontent.com/13151189/124797787-3fdca000-df85-11eb-87ac-8102c26f143c.png)
![image](https://user-images.githubusercontent.com/13151189/124797838-5125ac80-df85-11eb-9c41-5fccea954872.png)

**Comment by mrmlnc** [comment]
Yeeeeeeap. This is a bug introduced in the `3.2.6` version.

I don't remember why, but [it](https://github.com/mrmlnc/fast-glob/blob/d22afc5443d2d482fd494d6850939f2d0167e70e/src/managers/tasks.ts#L41) (an empty array for negative patterns) seemed like a good idea to me. Unfortunately, we didn't have tests for that. Now they will.

**Comment by longlho** [comment]
Thanks a lot for the quick fix! Can we get an emergency release for this fix please? 😄

**Comment by mrmlnc** [comment]
Must be fixed by [`3.2.7`](https://github.com/mrmlnc/fast-glob/releases/tag/3.2.7).
